src/renderer/base/renderer.hpp

```diff
@@ -90,10 +96,29 @@ namespace Microsoft::Console::Render
         static GridLineSet s_GetGridlines(const TextAttribute& textAttribute) noexcept;
         static bool s_IsSoftFontChar(const std::wstring_view& v, const size_t firstSoftFontChar, const size_t lastSoftFontChar);
 
+        // Base rendering loop
+        static DWORD WINAPI s_renderThread(void*) noexcept;
+        DWORD _renderThread() noexcept;
+        void _waitUntilCanRender() noexcept;
+
+        // Timer handling
+        void _starTimer(TimerHandle handle, TimerRepr delay, TimerRepr interval);
+        DWORD _calculateTimerMaxWait() noexcept;
+        void _waitUntilTimerOrRedraw() noexcept;
+        void _tickTimers() noexcept;
+        static TimerRepr _timerInstant() noexcept;
+        static TimerRepr _timerSaturatingAdd(TimerRepr a, TimerRepr b) noexcept;
+        static TimerRepr _timerSaturatingSub(TimerRepr a, TimerRepr b) noexcept;
+        static DWORD _timerToMillis(TimerRepr t) noexcept;
+
+        // Actual rendering
+        [[nodiscard]] HRESULT PaintFrame();

```

Comment:
It's a bit weird that this one is marked private, but doesn't have the `_` prefix. Plus there's also `_PaintFrame()` just below it.